File: internal/analyzer/testdata/python/good_patterns.py
from temporalio import workflow, activity
from datetime import timedelta


@workflow.defn
class GoodPatternWorkflow:
    @workflow.run
    async def run(self, input: str) -> str:
        result = await workflow.execute_activity(
            "my_activity",
            input,
            start_to_close_timeout=timedelta(seconds=30),
        )

        # Bounded loop with continue_as_new
        count = 0
        while True:
            await workflow.sleep(timedelta(seconds=1))
            count += 1
            if count > 100:
                await workflow.continue_as_new(input)

        return result


@activity.defn
async def good_activity(input: str) -> str:
    return input

File: internal/analyzer/testdata/java/GoodPatterns.java
package com.example.workflows;

import io.temporal.workflow.WorkflowInterface;
import io.temporal.workflow.WorkflowMethod;
import io.temporal.workflow.Workflow;
import io.temporal.activity.ActivityOptions;
import java.time.Duration;

@WorkflowInterface
public interface GoodPatternsWorkflow {
    @WorkflowMethod
    String run(String input);
}

public class GoodPatternsImpl implements GoodPatternsWorkflow {
    private final MyActivity activity = Workflow.newActivityStub(
        MyActivity.class,
        ActivityOptions.newBuilder()
            .withStartToCloseTimeout(Duration.ofSeconds(30))
            .build()
    );

    @WorkflowMethod
    public String run(String input) {
        activity.execute(input);
        while (true) {
            Workflow.sleep(Duration.ofSeconds(60));
            Workflow.continueAsNew(input);
        }
    }
}
